src/Core/Localization/CLDR/Reader.php

```diff
@@ -509,11 +516,36 @@ protected function mapLocaleData(SimpleXMLElement $xmlLocaleData, $supplementalD
 
         // Currencies
         $currencyData = $numbersData->currencies;
+        $currencyActiveDateThreshold = time() - self::CURRENCY_ACTIVE_DELAY * 86400;
         if (isset($currencyData->currency)) {
             $currencies = $localeData->getCurrencies();
             foreach ($currencyData->currency as $currencyNode) {
                 $currencyCode = (string) $currencyNode['type'];
 
+                // check if currency is still active in one territory
+                $isActiveSomewhere = false;
+                $currencyDates = $this->supplementalXml->supplementalData->xpath(
+                    '//region/currency[@iso4217="' . $currencyCode . '"]'
+                );
+                if (!empty($currencyDates)) {
+                    foreach ($currencyDates as $currencyDate) {
+                        if (empty($currencyDate->attributes()->to)) {
+                            // no date "to": currency is active in some territory
+                            $isActiveSomewhere = true;
```

Comment:
I don't know how many $currencyDate you may have, but having this part of the code in a separate method could allow you to return as soon as you can set the value as `true`